Refactor to use Web Audio API

hans00 opened the pull request:
Refactor `src/index.js` to use the Web Audio API instead of the `howler` library.

* Replace `howler` import with Web Audio API.
* Refactor `Sound` class to use Web Audio API for audio functionality.
* Update `Sound` class methods to use Web Audio API for play, pause, stop, volume control, and looping.
* Remove `howler` specific methods and properties.

Remove `howler` dependency from `package.json` and update `yarn.lock` accordingly.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/mybigday/react-native-web-sound/pull/1?shareId=6049d360-98cf-4e7c-827a-186f244dd9fa).

hans00 committed:
Refactor to use Web Audio API

Refactor `src/index.js` to use the Web Audio API instead of the `howler` library.

* Replace `howler` import with Web Audio API.
* Refactor `Sound` class to use Web Audio API for audio functionality.
* Update `Sound` class methods to use Web Audio API for play, pause, stop, volume control, and looping.
* Remove `howler` specific methods and properties.

Remove `howler` dependency from `package.json` and update `yarn.lock` accordingly.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/mybigday/react-native-web-sound?shareId=XXXX-XXXX-XXXX-XXXX).

hans00 commented:
@jhen0409 please take a look

jhen0409 commented on src/index.js:
```diff
@@ -15,27 +22,27 @@ Sound.setMode = () => {}
 Sound.prototype.pause = function() {
   if (!this.isLoaded()) return this
-  this.sound.pause()
+  this.source.stop(0)
```
Pause should be resumable, but I got error from pause+play:

```
Failed to execute 'start' on 'AudioBufferSourceNode': cannot call start more than once.
```

jhen0409 commented on src/index.js:
```diff
@@ -15,27 +22,27 @@ Sound.setMode = () => {}
   }
-  if (onEnd) this.sound.once('end', () => onEnd(true))
-  this.sound.play()
+  this.source.start(0)
```
Play should can be resumed from pause, so it shouldn't start with 0.

hans00 committed:
Replace `howler` with Web Audio API in `Sound` class

* **Remove `howler` dependency**
  - Remove `howler` from `package.json` and `yarn.lock`

* **Update `Sound` class**
  - Replace `howler` import with Web Audio API
  - Add property to track current playback position
  - Update `pause` method to store current playback position and use `suspend` method
  - Update `play` method to start audio from stored playback position and use `resume` method
  - Refactor methods to use Web Audio API for play, pause, stop, volume control, and looping

hans00 committed:
Correct implement

jhen0409 commented on src/index.js:
```diff
@@ -46,48 +60,54 @@ Sound.prototype.release = function() {
   return this
```
```suggestion
  this.audioContext.close()
  return this
```

jhen0409 committed:
Update src/index.js
